core/transaction.go

```diff
@@ -57,18 +58,20 @@ const (
 type ResourceBounds struct {
 	MaxAmount uint64
 	// MaxPricePerUnit is technically a uint128
-	MaxPricePerUnit *felt.Felt
+	MaxPricePerUnit *spec.Uint128
 }
 
 func (rb ResourceBounds) Bytes(resource Resource) []byte {
 	maxAmountBytes := make([]byte, 8) //nolint:mnd
 	binary.BigEndian.PutUint64(maxAmountBytes, rb.MaxAmount)
-	maxPriceBytes := rb.MaxPricePerUnit.Bytes()
+	maxPriceBytes := make([]byte, 16) //nolint:mnd
```

Comment:
I've seen this logic before (https://github.com/NethermindEth/juno/pull/2242/files#diff-8fcd72f4c5e1f056a6875680f66dfbfc8578908ff4fe889c8f1fc21e94e9c8bbR157) please DRY and create a function to do these steps, properly unittested :)

Reply:
Made changes